### ISSUE TITLE
French MP3 files and card previews are unintelligible

### ISSUE THREAD

**Issue by bbonb** [AwesomeTTS/AwesomeTTS#99]
Downloaded and started learning Anki about a week ago.  Using version 2.0.34 Qt 4.8.4 PyQt 4.10 on a Windows 10 machine.  Today installed AwesomeTTS and started trying to use it to provide French pronunciation for various words and phrases.  From the Anki browser, selected cards ("eight", "five") and used the AwesomeTTS  to configure service, enter French text to generate speech audio, and used Generate button to append to the French field of my card.  Tried the following service/voice combinations:   Acapela Group/Antoine,  Acapela Group/Claire, ImTranslator/Florence, Oddcast/Bernard.  In all cases, the Preview sound was reasonable, if not perfect.  But once I hit Generate, selecting the card and choosing Preview from the Browser window gives unintelligible audio.  I located the Anki folder containing sounds (...\Anki\User 1\collection.media) and played the generated MP3 files with Windows Media Player, and the MP3 files play the same unintelligible sounds provided by the Anki Browser preview.

My conclusions, so far:
1. All of the services tested are providing acceptable audio.
2. Something in the process of encoding or saving the audio to an MP3 file is changing the audio to make it unintelligible.
3. Since I have done nothing special, there must be a bug somewhere in the process.

I looked at the AwesomeTTS MP3 settings, but nothing is apparent to me.  Perhaps some flag to the LAME Transcoder, but I haven't touched those settings and wouldn't know what might be wrong.

**Comment by corpulentcoffee** [comment]
Hey there!

The files that get fed through on "Preview" should be the same physical files as get moved into your collection.

When you looked inside `Anki\User 1\collection.media`, how were those files named?

Cheers!
Dave

**Comment by corpulentcoffee** [comment]
(Also, if you want to post screenshots of your notes and templates, and also any other add-ons that you have installed, maybe I can figure out what is going on.)

**Comment by bbonb** [comment]
Hi Dave - thanks for your prompt response.
    
    Most recent examples if MP3 file names generated by AwesomeTTS:
        imtranslator-32d9ebe9-b9f127f8-b38b5e57-ffae999b-7efda9bb.mp3
        imtranslator-0875b8d1-e80143f7-8fb7a2b2-90f21bc6-81a2c513.mp3
    I expected it to be as you suggest.  These were generated with
    ImTranslator/Florence.
    
    I just ran it again, using Acapela Group/Antoine to generate the
    sound for "cinq":
        acapela-b4cdbb2b-c8874d33-59a89259-5fedf5f9-c1cd7f50.mp3
    I tried attaching it to a message at github, but apparently that
    won't come through, so I am attaching it here.
    
    I am also attaching 3 screen shots:  Anki Browser with the
    "five"/"cinq" card selected and displayed, with the above acapela
    MP3 inserted in the French field; Anki Template for the same card;
    AwesomeTTS dialog as used to generate the acapela MP3.  I don't see
    how the whole thing could be much simpler.  I haven't fiddled with
    any special features or options, either in Anki or AwesomeTTS,
    except to set the card type to include a reverse card.
    
    I hope you can listen to the MP3.  If it were correct, it should
    sound something like "sank" in English.  Instead it sounds similar
    in length and pitch to the original, but warped in some way that
    makes it completely unintelligible and thus useless.
    
    I hope this will mean something to you!
    
    Bruce
    
    On 4/4/2016 5:28 PM, Dave Shifflett
      wrote:
    
    
      Hey there!
      The files that get fed through on "Preview" should be the same
        physical files as get moved into your collection.
      When you looked inside Anki\User 1\collection.media,
        how were those files named?
      Cheers!
        Dave
      —
        You are receiving this because you authored the thread.
        Reply to this email directly or view
          it on GitHub

**Comment by corpulentcoffee** [comment]
Hi Bruce: things that you attach to e-mail won't come through to a GitHub issue, but you can e-mail me those screenshots /  MP3 samples directly at dave@corpulentcoffee.com and I will take a look.

Cheers!

**Comment by bbonb** [comment]
OK, today's behavior is different, but just as bad as yesterday's.  The first attached picture is a screenshot of the AwesomeTTS window as I used it to generate sound for a card whose English field contains "some, a few" and whose French field contains "quelques".  So the desire is that when the French text is displayed, a voice will speak "quelques".  When I click Preview in this window, the audio is as expected, a voice saying "quelques".

Then I click Generate and, after a delay, the Browser window shows the added sound, in the French field, as shown in the second screen shot.  Looks good, but when I click Preview in the Browser window, it displays correctly, but the audio is a French pronunciation of "some, a few".  Unlike yesterday, the audio from the Browser Preview is intelligible, it is just wrong!  The same sound is produced when I bring up the cards template (see final screen shot) and when I play the sound from the file with the Windows Media Player.

In other words, AwesomeTTS is grabbing the text from the source field (English) rather than the destination field (French), when it is time to actually generate a sound file.  I thought it was taking the text from the Preview box, but apparently not.  In fact, it is taking it from the wrong field.

Is it possible there is some sort of database issue that might be resolved by reinitialization?  If the text to be translated is coming from the wrong place, and a different wrong place yesterday from today, might this explain both faulty behaviors?

![ankiawesometts 160405](https://cloud.githubusercontent.com/assets/18273516/14303570/a11129fe-fb60-11e5-999c-20282b813340.jpg)
![ankibrowser 160405](https://cloud.githubusercontent.com/assets/18273516/14303569/a111047e-fb60-11e5-9687-cf6b2565ef53.jpg)
![ankitemplate 160405](https://cloud.githubusercontent.com/assets/18273516/14303568/a1102fea-fb60-11e5-8f63-08f426c4c1b0.jpg)

**Comment by corpulentcoffee** [comment]
Hi Bruce!

Two things:

- Based on your MP3s and your screenshots, it looks like you are feeding *English text* to a *French-speaking voice*. When you are in the "Add TTS Audio to Selected Notes" window, AwesomeTTS will go through each note you selected, take the text out of the **Source Field**, run it through your selected service, and save the MP3 output into your **Destination Field**... but you have your **Source Field** set to `English`. Because you *want* to read your French text from `French` *and also save the MP3 there*, you want to set **both** the **Source Field** and **Destination Field** to `French`.

- The "Add TTS Audio to Selected Notes" window is intended for adding audio to a large batch of notes. (That is why the "preview" sounds correct to you, as it's just there for testing the settings; because this window is intended for multiple notes, AwesomeTTS grabs the text for each card from that card rather than from the **Preview** field.) If you *do* want to do cards one at a time like that, where what you type into in the window is exactly what you'll get in the MP3s, you can use the ["Add TTS Audio to Note" window](https://ankiatts.appspot.com/usage/editor) instead by clicking the little speaker icon on the toolbar when you're in a note field.

Hope that helps!

**Comment by bbonb** [comment]
Dave, you're absolutely right!!

When I repeated my actions for which I wrote the original issue, having the translator pronounce "five" in French is pretty much unintelligible, until you realize what it is doing.  I thought "Source" and "Destination" simply referred to the first or question side of the flash card, and the second or answer side of the flash card, respectively.  A bit of documentation or help would have saved you the trouble of the investigation.  The user interfaces for both Anki and AwesomeTTS leave a lot to be desired.

Knowing the intent for this window also helps me understand, and I will likely use it for substantial batches of cards, as it was intended.

Thanks again for all your help.  I think I have it now, so will close the issue.
